src/broadcast/filelistener/filelistener.cpp

```diff
@@ -0,0 +1,101 @@
+
+#include <QTextCodec>
+#include <QtConcurrentRun>
+
+#include "broadcast/filelistener/filelistener.h"
+#include "broadcast/filelistener/metadatafileworker.h"
+#include "preferences/metadatafilesettings.h"
+
+
+FileListener::FileListener(UserSettingsPointer pConfig)
+        : m_COsettingsChanged(kFileSettingsChanged),
+          m_pConfig(pConfig),
+          m_latestSettings(MetadataFileSettings::getPersistedSettings(pConfig)) {
+
+    MetadataFileWorker *newWorker = new MetadataFileWorker(m_latestSettings.filePath);
+    newWorker->moveToThread(&m_workerThread);
+
+    connect(&m_workerThread,SIGNAL(finished()),
+            newWorker,SLOT(deleteLater()));
+
+    connect(this,SIGNAL(deleteFile()),
+            newWorker,SLOT(slotDeleteFile()));
+
+    connect(this,SIGNAL(moveFile(QString)),
+            newWorker,SLOT(slotMoveFile(QString)));
+
+    connect(this,SIGNAL(writeMetadataToFile(QByteArray)),
+            newWorker,SLOT(slotWriteMetadataToFile(QByteArray)));
+
+    connect(this,SIGNAL(clearFile()),
+            newWorker,SLOT(slotClearFile()));
+
+    connect(&m_COsettingsChanged,SIGNAL(valueChanged(double)),
+            this,SLOT(slotFileSettingsChanged(double)));
+
+    m_workerThread.start();
+}
+
+FileListener::~FileListener() {
+    m_workerThread.quit();
+    m_workerThread.wait();
+}
+
+
+void FileListener::slotBroadcastCurrentTrack(TrackPointer pTrack) {
+    if (!pTrack)
+        return;
+    m_fileContents.title = pTrack->getTitle();
+    m_fileContents.artist = pTrack->getArtist();
+    QString writtenString(m_latestSettings.fileFormatString);
+    writtenString.replace("$author",pTrack->getArtist()).
+            replace("$title",pTrack->getTitle()) += '\n';
+    QTextCodec *codec = QTextCodec::codecForName(m_latestSettings.fileEncoding);
+    DEBUG_ASSERT(codec);
```

Comment:
We should handle this gracefully. Eg use local8bit ()

Reply:
In reality it should never be null, but yeah, perhaps a warning and a fallback codec is better.

Reply:
We try to a avoid plain debug asserts, to not risk a crash in case we do a release build. Remember that this line is gone in release builds.

Reply:
It is changed by an if in the newest commit. Perhaps I should check all other DEBUG_ASSERTS

Reply:
use VERIFY_OR_DEBUG_ASSERT this is an if(!... in case of release version and an assert in case of debug build.

Reply:
Alright, thanks.